modules/obsidian/testbeds/obsidian.nix

```
@@ -0,0 +1,28 @@
+{ lib, pkgs, ... }:
+let
+  package = pkgs.obsidian;
+in
+{
+  nixpkgs.config.allowUnfreePredicate =
+    pkg:
+    builtins.elem (lib.getName pkg) [
+      "obsidian"
+    ];
+
+  stylix.testbed.ui.application = {
+    name = "obsidian";
+    inherit package;
+  };
+
+  home-manager.sharedModules = lib.singleton {
+    programs.obsidian = {
+      enable = true;
+      vaults.stylix = {
+        enable = true;
+        target = "stylix-vault";
+      };
+      inherit package;
+    };
+    stylix.targets.obsidian.vaultNames = [ "stylix" ];
+  };
```

From a quick test inside the testbed, it seems Obsidian creates the `$HOME/${valut}/`directory if `programs.obsidian.vaults.${vault}.target` is undeclared. For simplicity, it might be better to remove this declaration:

```suggestion
  home-manager.sharedModules = lib.singleton {
    programs.obsidian = {
      enable = true;
      vaults.stylix.enable = true;
      inherit package;
    };
    stylix.targets.obsidian.vaultNames = [ "stylix" ];
  };
```

What about creating a local `vault` variable to express the intent of some values being the same:

```suggestion
  home-manager.sharedModules =
    let
      vault = "stylix";
    in
    lib.singleton {
      programs.obsidian = {
        enable = true;
        vaults.${vault}.enable = true;
        inherit package;
      };

      stylix.targets.obsidian.vaultNames = [ vault ];
    };
```